smartsim/settings/base.py

```diff
@@ -156,7 +156,7 @@ def env_vars(self, value: t.Dict[str, t.Optional[str]]) -> None:
         self._env_vars = copy.deepcopy(value)
 
     # To be overwritten by subclasses. Set of reserved args a user cannot change
-    reserved_run_args = set()  # type: set[str]
+    reserved_run_args: frozenset[str] = frozenset()
```

Comment:
Without wanting to add too much additional work onto this PR, can we mark this as a `ClassVar`? Or do we actually treat it like an instance variable?

```suggestion
    reserved_run_args: t.ClassVar[frozenset[str]] = frozenset()
```

If this leads to additional typing errors, it might make sense to address them in a later PR, seeing as this is now, at best, tangential to upgrading to Python 3.12.

Reply:
Welcome to the co-authoring team!